.buildkite/scripts/run_scala_task.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+<<EOF
+Run a Scala task in Buildkite.  This includes:
+
+  - running tests
+  - building a Docker image (if applicable)
+  - publishing the Docker image to ECR (if on main)
```

Comment:
This seems like it does a lot more than just running a scala task! Can it be split up or renamed to make it clear what it does? Can the conditional (particularly re publishing on main) be put in the Buildkite pipeline instead? I know that the latter means the image needs to be rebuilt (as it's a new task) but I wonder if that tradeoff is OK.